msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -134,6 +134,13 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
 
             IdToken idToken = IdToken.Parse(response.IdToken);
 
+            //injecting tenantID into the IDToken if it is null
+            if (requestParams.Authority.AuthorityType == Core.Instance.AuthorityType.B2C)
```

Comment:
I'd wrap this up in a private method, smth like "InjectTenantIdB2C" or smth and add more details about B2C behaviour

Reply:
#Closed